File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] result = new int[2];

        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < numbers.length; i++){
            if(map.containsKey(numbers[i])){
                result = new int[]{map.getOrDefault(numbers[i],-1) + 1, i + 1};
                return result;
            }
            map.put(target - numbers[i], i);
        }

        return result;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Optimized solution
        nums_map = {}
        result = []
        for i in range(len(nums)):
            if target - nums[i] in nums_map:
                result.append(nums_map[target-nums[i]])
                result.append(i)
                return result
            nums_map[nums[i]] = i
        return result
        